blog/2023-11-27-magazine-006.md

```diff
@@ -0,0 +1,195 @@
+hide_table_of_contents: true
+---
+> The 6th issue of the Wing Inflight Magazine.
+> <!--truncate-->
+
+Hey folks!
+
+We’re back with another issue of the **Wing Inflight Magazine** and we are excited to share some updates about [winglang](https://winglang.io).
+
+- 🚀 [New: bring ui](#bring-ui)
+- 🚀 [New: Singleton](#singleton)
+- 🔥 [Remove $ from String Interpolations](#remove--from-string-interpolations)
+- 🚀 [New: Expose AWS Resources](#expose-aws-resources)
+- 🚀 [New: Sourcemaps and Stacktraces](#sourcemaps-and-stacktraces)
+- 📅 [Upcoming Events](#upcoming-events)
+- 🎥 [The Wingly Update](#the-wingly-update)
+
+### bring ui
+
+Introducing the ui module to the SDK, a collection of APIs for adding visual interactions for your classes in the Wing Console. UI components aren't displayed in the Wing Console quite yet, but they can be accessed and called through the simulator API!
+
+```js
+bring cloud;
+bring ui;
+
+class WidgetService {
+  data: cloud.Bucket;
+  counter: cloud.Counter;
+  new() {
+    this.data = new cloud.Bucket();
+    this.counter = new cloud.Counter();
+    
+    // a button lets you invoke any inflight function
+    new ui.Button("Add widget", inflight () => { this.addWidget(); });
+    
+    // a field displays a labeled value, with optional refreshing
+    new ui.Field(
+      "Total widgets",
+      inflight () => { return this.countWidgets(); },
+      refreshRate: 5s,
+    );
+  }
+
+  inflight addWidget() {
+    let id = this.counter.inc();
+    this.data.put("widget-${id}", "my data");
+  }
+
+  inflight countWidgets(): str {
+    return "${this.data.list().length}";
+  }
+}
+
+new WidgetService();
+```
+
+❤️ [Chris Rybicki](https://github.com/Chriscbr) ❤️
+
+### Singleton
+
+If you want to create one instance of a resource thats the way to go!
+We are not making it easy to access the root app from *every* node through `Node.of(this).app` or `Node.of(this).root`. Both return the same root app, even if it's not the absolute root of the tree. `tryFindChild(id)` is also added to the app to make it easier to implement app-level singletons!
+
+A few use cases within Wing for a singleton include: When adding a provider and ensuring the singular presence of it. Another scenario involves implementing diverse checks for storing resources in a unified global bucket, with the aim of maintaining a singular presence.
+
+Check this out:
+
+```js
+
+ class SingletonBucket {
+   pub static of(scope: std.IResource): cloud.Bucket {
+     let uid = "SingletonBucket";
+     let root = std.Node.of(scope).root;
+     return unsafeCast(root.tryFindChild(uid)) ?? new cloud.Bucket() as uid in root;
+   }
+ }
+
+ let q1 = new cloud.Queue() as "q1";
+ let store1 = SingletonBucket.of(q1);
+
+ test "singleton" {
+   store1.put("hello", "world");
+ }
+```
+
+❤️ [Elad Ben-Israel](https://github.com/eladb) ❤️
+
+
+### Remove `$` from `String` Interpolations
+
+#### This one is a life changer :)
```

Comment:
for me its a change for the worse :)

Reply:
You have to be positive about it... fake it until you make it :)